lib/logstash/inputs/kafka.rb

```diff
@@ -92,7 +92,7 @@ class LogStash::Inputs::Kafka < LogStash::Inputs::Base
   # IP addresses for a hostname, they will all be attempted to connect to before failing the 
   # connection. If the value is `resolve_canonical_bootstrap_servers_only` each entry will be 
   # resolved and expanded into a list of canonical names.
-  config :client_dns_lookup, :validate => ["default", "use_all_dns_ips", "resolve_canonical_bootstrap_servers_only"], :default => "default"
+  config :client_dns_lookup, :validate => ["use_all_dns_ips", "resolve_canonical_bootstrap_servers_only"], :default => "use_all_dns_ips"
```

Comment:
What did `default` previously map to in Kafka?

By not allowing the setting of `default`, this is potentially a breaking change (even though it is somewhat unlikely that a user will explicitly set `default`, it's a valid setting). If we choose to remove the explicit `default` value we need to flag it in change logs and release notes, and figure out how we choose to bundle this with future logstashes. We would also need to update the docs for the `input` and `output`

Reply:
I agree with you on the breaking change. Given the explanation in [previous comment](https://github.com/logstash-plugins/logstash-integration-kafka/pull/130#discussion_r1047398099) do you think that maybe is better if we implicitly remap `default` to `use_all_dns_ips`?. So that if one user has explicitly set `default` we switch implicitly to `use_all_dns_ips`. In worst case we let them to connect to another node while the previous setting would have resulted in connection error.